packages/grid/src/grid.ts

```diff
@@ -37,14 +62,57 @@ export class Grid<T extends { [x: string]: unknown } = Record<string, unknown>>
   renderItem(item: T): TemplateResult {
     return html`
       <tr>
-        ${this.columns.map(col => col.renderContentCell(item))}
+        ${this.columns.map(col => col.renderData(item))}
       </tr>
     `;
   }
 
+  /** Updates the `width` of all columns which have `autoWidth` set to `true`. */
+  async recalculateColumnWidths(): Promise<void> {
+    // Do not remove, this is needed; not sure why
+    await this.updateComplete;
+
+    this.columns
+      .filter(col => !col.hidden && col.autoWidth)
+      .forEach(col => {
+        const index = this.columns.indexOf(col),
+          cells = this.renderRoot.querySelectorAll<HTMLElement>(`:where(td, th):nth-child(${index + 1})`);
+
+        col.width = Array.from(cells).reduce((acc, cur) => {
+          cur.style.flexGrow = '0';
+          cur.style.width = 'auto';
+
+          const { width } = cur.getBoundingClientRect();
+
+          cur.style.flexGrow = cur.style.width = '';
+
+          return Math.max(acc, width);
+        }, 0);
+      });
+
+    this.requestUpdate('columns');
+  }
+
   #onSlotchange(event: Event & { target: HTMLSlotElement }): void {
```

Comment:
minor change :) maybe `#onSlotChange` instead of `#onSlotchange`

Reply:
Well the name of the DOM event is `slotchange`, so that's why `#onSlotchange`